types/types.go

```diff
@@ -486,6 +484,7 @@ type SystemContext struct {
 	// Allow contacting docker registries over HTTP, or HTTPS with failed TLS verification. Note that this does not affect other TLS connections.
 	DockerInsecureSkipTLSVerify OptionalBool
 	// if nil, the library tries to parse ~/.docker/config.json to retrieve credentials
+	// the value of Username and Password in DockerAuthConfig can be empty
```

Comment:
This would be better placed as a comment in the `DockerAuthConfig` structure, and it should also say what _does it mean_ for the values to be empty.

Reply:
fixed ✋